ggshield/core/config.py

```diff
@@ -332,22 +307,8 @@ def __init__(self, instance: str):
 
 @dataclass
 class AuthConfig(YAMLFileConfig):
-    current_token: Optional[str] = None
-    current_instance: Optional[str] = None
-    default_instance: str = DEFAULT_DASHBOARD_URL
-    default_token_lifetime: Optional[int] = None
```

Comment:
Because it's never used: the OAuth code uses the value from InstanceConfig.

Reply:
@agateau-gg That's a problem, the default_token_lifetime is useful. But it cannot be used as you say because a None value in the instanceConfig will be considered as a "no expiry". 

I propose to use 0 as "no expiry" and disallowing `None` for default_lifetime_config, but allowing it for the instance Config lifetime.